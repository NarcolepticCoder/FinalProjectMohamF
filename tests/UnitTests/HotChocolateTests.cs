using Data;
using Data.Entities;
using GraphQL;
using GraphQL.Mutations;
using GraphQL.Repositories;
using HotChocolate;
using HotChocolate.Execution;
using Microsoft.EntityFrameworkCore;
using Microsoft.Extensions.DependencyInjection;
using Microsoft.VisualBasic;
using Snapshooter.Xunit;
using UnitTests.Helpers;

namespace UnitTests
{
    public class HotChocolateTests
    {
        [Fact]

        public async Task QueryShouldReturnListOfUsers()
        {
            

            var executor = await TestSchemaBuilder.BuildAsync(seed: db =>
            {

                db.Users.Add(new User
                {
                    Id = Guid.Parse("aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa"),
                    Email = "test123@shouldwork.com",
                    ExternalId = "123",
                    Role = new Roles { Id = Guid.Parse("cccccccc-cccc-cccc-cccc-cccccccccccc"), Name = "Mo" },
                });
                db.Users.Add(new User
                {
                    Id = Guid.Parse("bbbbbbbb-bbbb-bbbb-bbbb-bbbbbbbbbbbb"),
                    Email = "test321@shouldwork.com",
                    ExternalId = "321",
                    Role = new Roles { Id = Guid.Parse("dddddddd-dddd-dddd-dddd-dddddddddddd"), Name = "Works" },
                });


            });


            var result = await executor.ExecuteAsync(@"
            query {
            users {
            id
            email
            role {
                id
                name
                  }
                }
              }
            ");
            
            
            result.ToJson().MatchSnapshot();
        }
        
        [Fact]
        public async Task SchemaTest()
        {
            var executor = await TestSchemaBuilder.BuildAsync();

            //print schema from executor
            var sdl = executor.Schema.Print();

            sdl.ToString().MatchSnapshot();
            
            
        }
    }
}