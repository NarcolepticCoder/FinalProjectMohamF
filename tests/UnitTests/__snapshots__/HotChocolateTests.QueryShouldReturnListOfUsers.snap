{
  "data": {
    "users": [
      {
        "id": "aaaaaaaa-aaaa-aaaa-aaaa-aaaaaaaaaaaa",
        "email": "test123@shouldwork.com",
        "role": {
          "id": "cccccccc-cccc-cccc-cccc-cccccccccccc",
          "name": "Mo"
        }
      },
      {
        "id": "bbbbbbbb-bbbb-bbbb-bbbb-bbbbbbbbbbbb",
        "email": "test321@shouldwork.com",
        "role": {
          "id": "dddddddd-dddd-dddd-dddd-dddddddddddd",
          "name": "Works"
        }
      }
    ]
  }
}
